backend/LeetWars.Core/LeetWars.Core.BLL/Services/PreferencesService.cs

```diff
@@ -0,0 +1,71 @@
+﻿using AutoMapper;
+using LeetWars.Core.BLL.Exceptions;
+using LeetWars.Core.BLL.Interfaces;
+using LeetWars.Core.Common.DTO.UserPrefferences;
+using LeetWars.Core.DAL.Context;
+using LeetWars.Core.DAL.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace LeetWars.Core.BLL.Services
+{
+    public class PreferencesService : BaseService, IPreferencesService
+    {
+        private readonly IUserGetter _userGetter;
+        public PreferencesService(LeetWarsCoreContext context,
+                       IMapper mapper,
+                       IUserGetter userGetter
+                       ) : base(context, mapper)
+        {
+            _userGetter = userGetter;
+        }
+        public async Task<UserPreferencesDto> GetUserPreferences()
+        {
+            var currentUser = await _context.Users.FirstOrDefaultAsync(x => x.Uid == _userGetter.CurrentUserId)
+                             ?? throw new NotFoundException(nameof(User), _userGetter.CurrentUserId);
+
+            var preferences = await _context.UserPreferences
+                .Include(l => l.Language)
+                .FirstOrDefaultAsync(x => x.UserId == currentUser.Id);
+
+            return _mapper.Map<UserPreferencesDto>(preferences);
+        }
+
+        public async Task<UserPreferencesDto> SetUserPreferences(NewUserPreferencesDto newPreferences)
+        {
+            var currentUser = await _context.Users.FirstOrDefaultAsync(x => x.Uid == _userGetter.CurrentUserId)
```

Comment:
the same. it would be better to have UserId in _userGetter

Reply:
_userGetter.CurrentUserId returned User Uid, not fusr user Id
but i should use _userGetter.GetCurrentUserOrThrow(); method